I_budget_layout_bauen

joeymez opened the pull request:
Layout
- Hinzufügen einer neuen Budget-Anzeige (Kreisform mit Prozentanzeige)
- Hinzufügen eines AlterDialog-Fensters mit vier Eingabefeldern

Logik
- Implementierung der Logik für AlertDialog-Fenster (Eingabe werden in die CardView übergeben)
- Implementierung der Logik zum Befüllen der CardView (wird durch die Übergaben erstellt)

Anpassung
- Formatierung des neuen Layouts
- Formatierung des Codes & Kommentieren

Images
- Hinzufügen von vier neuen Images für das AlertDialog-Fenster
- Einfügen dieser in das PopUp-Fenster

Drawable
- Hinzufügen zweier neue XML für die Kreisanzeige in der CardView

HINWEIS: Ein kleiner Bug ist drin  ... es handelt sich hierbei um das eingeben eines Budgetbetrages. Es können derzeit nur Ganze Zahlen eingegeben werden. Diesen Bug BITTE in den nächsten Sprint ins e_bug ;)

Commit:
VORLÄUFIGE Version i_budget_layout_bauen:
--------------------------------------------------

- Einfügen einer CardView in das Layout des Budget
- Erster Versuche diese in eine RecyclerView zu implementieren

WICHTIG: Code unvollständig und somit NICHT FERTIG!!! ;)

SshaBrn committed:
Recyclerview mit Cardviews testweise erstellt.

- C_Budget_CardView_Adapter

erster Test. Layout muss überarbeitet werden.

Commit:
Update / Verbesserung des Items i_budget_layout_bauen:
---------------------------------------------------------------

- Komplette Überarbeitung und Neugestaltgung des Layouts
- Hinzufügen der CardView in die RecyclerView
- Hinzufügen von TEST-DATEN (RANDOM-Daten) zur Befüllung der CardView
- Hinzufügen der AppBar-Leiste mit Back-Button
- Hinzufügen neuer Icons für die spätere Verwendung
- Hinzufügen der Logik für die Prozessbarleiste

Commit:
Update / Verbesserung des Items i_budget_layout_bauen:
---------------------------------------------------------------

- Beim drücken des FAB wird ein neuer Item erzeugt (völlige random Daten) :)

Commit:
Update / Verbesserung des Items i_budget_layout_bauen:
---------------------------------------------------------------

- Hinzufügen der Icons ins Git

Commit:
Update / Verbesserung des itmes i_budget_layout_bauen:
---------------------------------------------------------------

- Hinzufügen neuer Funktionen zur Verwaltung der CardView / RecyclerView
- Swipper-Funktion neu implementiert (Löschen der Cards durch swippen)
- Hinzufügen eines Toast-Fensters bei löschen

Commit:
Update / Verbesserung des itmes i_budget_layout_bauen:
---------------------------------------------------------------

- Alert Dialog hinzugefügt

Commit:
Fertigstellung des Items i_budget_layout_bauen:
------------------------------------------ ----------

Layout
- Hinzufügen einer neuen Budget-Anzeige (Kreisform mit Prozentanzeige)
- Hinzufügen eines AlterDialog-Fensters mit vier Eingabefeldern

Logik
- Implementierung der Logik für AlertDialog-Fenster (Eingabe werden in die CardView übergeben)
- Implementierung der Logik zum Befüllen der CardView (wird durch die Übergaben erstellt)

Anpassung
- Formatierung des neuen Layouts
- Formatierung des Codes & Kommentieren

Images
- Hinzufügen von vier neuen Images für das AlertDialog-Fenster
- Einfügen dieser in das PopUp-Fenster

Drawable
- Hinzufügen zweier neue XML für die Kreisanzeige in der CardView

Commit:
Kleiner Formatänderung ... nichts großes :)

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+                    public void onClick(DialogInterface dialog, int which) {
+
+                        // CARD VIEW wird befüllt wenn auf OK gedrückt wird
+                        addItem(budgetContent.getText().toString(),
```
Wie schon mit dem Johanns besprochen. Bitte noch prüfen ob die Felder null oder isEmpty() sind sonst stürtzt die App ab wenn man ohne Eingabe auf OK drückt.

dontcallme replied:
App stürzt bei leeren feldern ab. Wie domi schon sagte bitte auf Null/Empty checken

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+                        // CARD VIEW wird befüllt wenn auf OK gedrückt wird
+                        addItem(budgetContent.getText().toString(),
+                                budgetContent.getText().toString(),
+                                monthContent.getText().toString(),
```
Meiner Meinung nach wäre es an der Stelle statt Monat und Jahr explizit eingeben zu können besser wenn man ein "Von" Datum und ein "Bis" Datum auswählen kann (eventuell über einen Datepicker). 
Ich weiß das ist zu kurzfristig, daher könnten wir ja eventuell ein weiteres Item zur Verbesserung der Maske wie es bei der Manuellen auch gemacht wurde in den nächsten Sprint aufnehmen.

dontcallme replied:
Eine von/bis angabe wäre auch mein Vorschöag. A besten mit 2 datepickern. 
Ohne Datepicker müsste man sich um eine Formatierung kümmern, damit alles einheitlich aussieht.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+    // ---------------------------------- ITEM SWIPPER BEGINN -----------------------------------
+
+    // CREATE HELPER CALLBACK - Funktion zum Swippen (bewegen od. löschen von Items durch swippen)
+    private ItemTouchHelper.Callback createHelperCallBack(){
```
Bitte an die allgemeine Dokumentationsform halten. Also:
- Methodenkommentar (über der Methode): /**  KOMMENTAR */
- Zeilenkommentar (neben der Zeile) // KOMMENTAR

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+
+
+    // MOVE ITEM - Bewegen eines Items in der RecyclerView (Swip nach Oben oder Unten)
+    private void moveItem(int oldPos, int newPos){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+
+
+    // DELETE ITEM - Löschen eines Items in der RecyclerView (Swip nach Links)
+    private void deleteItem(int position){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+
+
+    // ADD ITEM - Befüllung der RecyclerView mit EINER Datenmenge
+    private void addItem(String budgetMax, String budgetCurrently, String turnus, String title, String year){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+    // ------------------------------------- TESTDATEN BEGINN ------------------------------------
+
+    // RANDOM NUMBER - Zufällige Auswahl eines Int zwischen Max und Min
+    public int randomNumber(int max, int min){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+
+
+    // RANDOM TITLE - Zufällige Auswahl eines Titels
+    public String randomTitle(int random){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+
+
+    // RANDOM MONTH - Zufällige Auswahl eines Monats
+    public String randomMonth (int random){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/A_Budget.java:
```diff
@@ -20,16 +39,233 @@ protected void onCreate(Bundle savedInstanceState) {
+
+
+    // CALCULATE PERCANTAGE - Berechnen des Prozentsatzes
+    private int percentageCalculator(int max, int currently){
```
Wie oben beschrieben, Kommentare bitte anpassen.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+        this.year = year;
+    }
+
+    public int getBudgetMax() {
```
Bitte auch die Getter und Setter Dokumentieren (wie oben beschrieben)

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+        return budgetMax;
+    }
+
+    public String getTurnus() {
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+        return turnus;
+    }
+
+    public void setTurnus(String turnus) {
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+        this.turnus = turnus;
+    }
+
+    public int getProcessbar() {return processbar;    }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+
+    public int getProcessbar() {return processbar;    }
+
+    public void setProcessbar(int processbar) {
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+        this.processbar = processbar;
+    }
+
+    public void setBudget(int budget) {
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+        this.budgetMax = budget;
+    }
+
+    public String getTitle() {        return title;    }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+
+    public String getTitle() {        return title;    }
+
+    public void setTitle(String title) {        this.title = title;    }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+    public void setTitle(String title) {        this.title = title;    }
+
+
+    public int getBudgetCurrently() {        return budgetCurrently;    }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+
+    public int getBudgetCurrently() {        return budgetCurrently;    }
+
+    public void setBudgetCurrently(int budgetCurrently) {        this.budgetCurrently = budgetCurrently;    }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+
+    public void setBudgetCurrently(int budgetCurrently) {        this.budgetCurrently = budgetCurrently;    }
+
+    public int getProcessPercentage() {        return (int)(budgetCurrently*100/budgetMax);   }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+
+    public int getProcessPercentage() {        return (int)(budgetCurrently*100/budgetMax);   }
+
+    public String getYear() {return year;    }
```
Wie gerade beschrieben.

schmitzdominic commented on app/src/main/java/de/projektss17/bonpix/daten/C_Budget.java:
```diff
@@ -0,0 +1,62 @@
+
+    public String getYear() {return year;    }
+
+    public void setYear(String year) {        this.year = year;    }
```
Wie gerade beschrieben.

schmitzdominic reviewed (changes requested):
Im großen und ganzen sehr gut 👍 
Mir gefällt die Maske ziemlich gut auch vom Design her.
Bitte nur noch
- Kommentare ändern
- Prüfen ob die Felder leer

dontcallme reviewed (changes requested):
Bitte die im Code angemerkten Punkte noch ändern. (Reply bei den Anmerkungen vom Domi)

Ansonsten passt alles, sieht gut aus :)

Commit:
Update Item i_budget_layout_bauen:

- Standardisierung der Kommentierung
- Abfangen der Exception wenn Inhalt nicht befüllt wird (Dialog-Fenster schließt einfach)

HINWEIS: Alle anderen Änderungswünsche (z.B. Monats- & Jahreseingabe) wird im nächsten Sprint erfolgen!

Commit:
Update Item i_budget_layout_bauen:

- Standardisierung der Kommentierung (C_Budget)

dontcallme reviewed (approved):
Von mir gibts ein OK, App stürtzt nicht mehr ab bei fehlenden feldern.

Gut gemacht :)

schmitzdominic reviewed (approved):
Gut gemacht ;) APPROVED
